LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DefaultLoRaDataRequestHandler.cs

```diff
@@ -212,7 +212,7 @@ public async Task<LoRaDeviceRequestProcessResult> ProcessRequestAsync(LoRaReques
                     {
                         try
                         {
-                            decryptedPayloadData = loraPayload.GetDecryptedPayload(loRaDevice.AppSKey);
+                            decryptedPayloadData = loraPayload.GetDecryptedPayload(loRaDevice.AppSKey.Value);
```

Comment:
The code scanning is right. The AppSKey, might be null?

Reply:
Yes, I have that in other places as well so might as well track the whole discussion here. I think that's a direct consequence of not having two different types for `ABP`/`OTAA` devices and incrementally updating properties on these devices after creation.

My thinking was that we have tests to verify what our types do not guarantee - in this case that the value is not `null`. I could add a manual check and throw an exception with a custom message if the value is `null`, but at the end of the day that's more or less the same as the default exception that gets thrown by accessing `.Value` on a `null`.

Reply:
We had a discussion on this. We agree the proper fix would be to have dedicated types, but that would be a very large change. Hence, we rely on the tests to guarantee the key won't be null here.